lib/std/os.zig

```diff
@@ -438,6 +438,39 @@ pub fn pread(fd: fd_t, buf: []u8, offset: u64) PReadError!usize {
     return index;
 }
 
+pub const TruncateError = error{
+    /// The file descriptor is not open for writing.
+    NotFile,
+} || UnexpectedError;
+
+pub fn truncate(fd: fd_t, length: u64) TruncateError!void {
```

Comment:
k.

> (actually looks like it might be a typo based on the rest of the PR)

I didn't like `ftruncate` and wasn't planning on adding `truncate` in `os.zig` so I went for a nicer name.